psd-web/spec/features/password_reset_spec.rb

```diff
@@ -0,0 +1,124 @@
+require "rails_helper"
+
+RSpec.describe "Password reset management", :with_test_queue_adpater do
+  let(:user)                         { create(:user) }
+  let!(:reset_token)                 { Devise.token_generator.generate(User, :reset_password_token) }
+  let(:edit_user_password_url_token) { edit_user_password_url(reset_password_token: reset_token.first) }
+
+
+  def send_reset_password
+    allow(Devise.token_generator)
+      .to receive(:generate)
+            .with(User, :reset_password_token).and_return(reset_token)
+    _raw, enc = reset_token
+    user.update!(reset_password_token: enc)
+
+    visit new_user_session_path
+
+    click_link "Forgot your password?"
+
+    perform_enqueued_jobs do
+      body = {
+        email_address: user.email,
+        template_id: NotifyMailer::TEMPLATES[:reset_password_instruction],
+        reference: "Password reset",
+        personalisation: {
+          name: user.name,
+          edit_user_password_url_token: edit_user_password_url_token
+        }
+      }
+
+      stub_request(:post, "https://api.notifications.service.gov.uk/v2/notifications/email")
+        .with(body: body.to_json).to_return(status: 200, body: {}.to_json, headers: {})
+
+      fill_in "Email address", with: user.email
+      click_on "Send email"
+    end
+  end
+
+  context "when entering an invalid email" do
+    it "does not allow you to reset you pasword" do
+      visit new_user_session_path
+
+      click_link "Forgot your password?"
+      fill_in "Email address", with: "not_an_email"
+      click_on "Send email"
+
+      expect(page).to have_css("h2#error-summary-title", text: "There is a problem")
+      expect(page).to have_link("Enter your email address in the correct format, like name@example.com", href: "#email")
+    end
+  end
+
+  context "with a valid token" do
+    it "does not allow you to reset you pasword" do
+      send_reset_password
+      expect(page).to have_css("p.govuk-body", text: "Click the link in the email to reset your password.")
```

Comment:
I think we could probably remove the assertion about the css here:

```suggestion
      expect(page).to have_content("Click the link in the email to reset your password.")
```

Reply:
We've used `have_css` everywhere so far, it also a documents directly how this piece of text is displayed. I find is easier to reason about.